### PR TITLE
ignition.dsl: add brew install bottle jobs

### PR DESCRIPTION
**Pull request by scpeters**
This adds jobs for testing installation of homebrew bottles for supported ignition versions. I generated the jobs with the following dsl build:

* https://build.osrfoundation.org/job/_dsl_ignition/976/

~~~
Added items:
    GeneratedJob{name='ignition_cmake1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_cmake2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_common1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_common2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_common3-install-homebrew-bottle'}
    GeneratedJob{name='ignition_fuel-tools1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_fuel-tools2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_fuel-tools3-install-homebrew-bottle'}
    GeneratedJob{name='ignition_fuel-tools4-install-homebrew-bottle'}
    GeneratedJob{name='ignition_gazebo2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_gazebo3-install-homebrew-bottle'}
    GeneratedJob{name='ignition_gui0-install-homebrew-bottle'}
    GeneratedJob{name='ignition_gui2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_gui3-install-homebrew-bottle'}
    GeneratedJob{name='ignition_launch1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_launch2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_math2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_math4-install-homebrew-bottle'}
    GeneratedJob{name='ignition_math5-install-homebrew-bottle'}
    GeneratedJob{name='ignition_math6-install-homebrew-bottle'}
    GeneratedJob{name='ignition_msgs1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_msgs2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_msgs4-install-homebrew-bottle'}
    GeneratedJob{name='ignition_msgs5-install-homebrew-bottle'}
    GeneratedJob{name='ignition_physics1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_physics2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_plugin0-install-homebrew-bottle'}
    GeneratedJob{name='ignition_plugin1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_rendering0-install-homebrew-bottle'}
    GeneratedJob{name='ignition_rendering2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_rendering3-install-homebrew-bottle'}
    GeneratedJob{name='ignition_rndf-install-homebrew-bottle'}
    GeneratedJob{name='ignition_sensors2-install-homebrew-bottle'}
    GeneratedJob{name='ignition_sensors3-install-homebrew-bottle'}
    GeneratedJob{name='ignition_tools0-install-homebrew-bottle'}
    GeneratedJob{name='ignition_tools1-install-homebrew-bottle'}
    GeneratedJob{name='ignition_transport4-install-homebrew-bottle'}
    GeneratedJob{name='ignition_transport5-install-homebrew-bottle'}
    GeneratedJob{name='ignition_transport7-install-homebrew-bottle'}
    GeneratedJob{name='ignition_transport8-install-homebrew-bottle'}
~~~

* [Build Status](https://build.osrfoundation.org/job/ignition_cmake2-install-homebrew-bottle/1/) https://build.osrfoundation.org/job/ignition_cmake2-install-homebrew-bottle/1/
* [Build Status](https://build.osrfoundation.org/job/ignition_common1-install-homebrew-bottle/3/) https://build.osrfoundation.org/job/ignition_common1-install-homebrew-bottle/3/
* [Build Status](https://build.osrfoundation.org/job/ignition_launch1-install-homebrew-bottle/3/) https://build.osrfoundation.org/job/ignition_launch1-install-homebrew-bottle/3/

### CODE REVIEW

**Comment by scpeters** [comment]
changed logic in 8a720a0 to mark builds as unstable if the formula doesn't have a  test defined:

* [Build Status](https://build.osrfoundation.org/job/ignition_launch1-install-homebrew-bottle/4/) https://build.osrfoundation.org/job/ignition_launch1-install-homebrew-bottle/4/

File: jenkins-scripts/dsl/ignition.dsl [hunk 1]
```diff
@@ -553,6 +553,29 @@ ignition_software.each { ign_sw ->
         }
     }
   }
+
+  // 3. install jobs to test bottles
+  supported_install_pkg_branches(ign_sw).each { major_version, supported_distros ->
+    def install_default_job = job("ignition_${ign_sw}${major_version}-install-homebrew-bottle")
```
Comment (j-rivero) [inline jenkins-scripts/dsl/ignition.dsl:559]: Although we don't have a clear policy on job naming, my intention has been to reserve the last post-fixes to indicate platform. We can include the bottle information together with the install word, something like: 

```suggestion
    def install_default_job = job("ignition_${ign_sw}${major_version}-install_bottle-homebrew-amd64")    
```

Comment (scpeters) [reply jenkins-scripts/dsl/ignition.dsl]: regenerated bottle jobs with new names in https://build.osrfoundation.org/job/_dsl_ignition/977/

**Comment by scpeters** [comment]
> The PR does not implement any kind of Mac version restriction to test bottles in. I think that we are probably restricting bottle testing to Mojave and happen that we probably support it in the vast majority of bottles but we might need to fix that in the future.

It's currently inheriting from `OSRFOsXBase`, which applies the label `osx`, which is currently given to machines using `mojave`.

* https://github.com/ignition-tooling/release-tools/blob/master/jenkins-scripts/dsl/_configs_/OSRFOsXBase.groovy#L20

So yes, this will not test any high-sierra bottles; in the future we should ensure that we test each version of our supported bottles, but I'd like to save that for the future.

**Comment by scpeters** [comment]
`ignition-launch*` formulae don't have tests, so they are marked as unstable:

* [Build Status](https://build.osrfoundation.org/job/ignition_launch1-install_bottle-homebrew-amd64/4/) https://build.osrfoundation.org/job/ignition_launch1-install_bottle-homebrew-amd64/4/

**Comment by scpeters** [comment]
I'll go ahead and merge this so the install builds stop failing

**Pull request merged**